docker/debug.Dockerfile

```diff
@@ -0,0 +1,18 @@
+# Copyright (c) 2023 -  Restate Software, Inc., Restate GmbH.
+# All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# This docker file is mainly used to build a local docker image
+# with restate binary to run sdk tests
+
+FROM archlinux/archlinux AS runtime
```

Comment:
arch linux usually have the most uptodate libc version. hence using this as a base will have a higher chance of supporting a locally built binary.

Of course this is only helpful for linux users (so far). I can look later into Mac support.